### ISSUE TITLE
Mail calls non-existent Net::SMTP disable_starttls_auto

### ISSUE THREAD

**Issue by vladimir-mencl-eresearch** [mikel/mail#1571]
Hi,

After updating to mail 2.8.0, mail delivery from my app stopped working for me.

I have `enable_starttls_auto: false` in my Mail `delivery_method` defaults.

The recent changes to `lib/mail/network/delivery_methods/smtp.rb` (d91f335) call `disable_starttls_auto` method that does not exist in `net/smtp.rb`, resulting into:
```
2023-03-20 10:43:04 - NoMethodError - undefined method `disable_starttls_auto' for #<Net::SMTP localhost:25 started=false>
```

I believe this could be easily fixed by replacing the `disable_starttls_auto` call with `disable_starttls` - looking at https://github.com/ruby/net-smtp/blob/master/lib/net/smtp.rb, the "auto" behaviour is just tracked by setting `@starttls` to `:auto`, and setting it to `false` would do the right thing.

Cheers,
Vlad

**Comment by vladimir-mencl-eresearch** [comment]
FYI @jeremy @eval

**Comment by eval** [comment]
@vladimir-mencl-eresearch this was fixed in v2.8.1.

**Comment by vladimir-mencl-eresearch** [comment]
Ah, I now see it was - should, should have checked before reporting.
